RELEASE.md

```diff
@@ -0,0 +1,43 @@
+# spin-operator release process
+
+The vast majority of the release process is handled after you push a new git tag.
+
+First, let's start by setting some environment variables we can reference later.
+
+```console
+export TAG=v0.x.0 #CHANGEME
+```
+
+To push a tag, do the following:
+
+```console
+git checkout main
+git remote add upstream git@github.com:spinkube/spin-operator
+git pull upstream main
+git tag --sign $TAG --message "Release $TAG"
+git push upstream $TAG
+```
+
+Observe that the [CI run for that tag](https://github.com/spinkube/spin-operator/actions) completed.
+
+Bump the Helm chart versions. See #311 for an example.
+
+Next, you'll need to update the documentation:
+
+```console
+git clone git@github.com:spinkube/documentation
+cd documentation
+```
+
+Change all references to the previous version to the new version, then run:
+
+```console
+npm install
+npm run generate:crd:reference
```

Comment:
I don't think we actually want to do this. It seems like we don't check in the reference, we just rely on the CI to generate it. It's just important that we bump the version in that generate script.